src/creator.rs

```diff
@@ -54,8 +53,12 @@ impl<H: Hasher> Creator<H> {
         }
     }
 
-    // initializes the vectors corresponding to the given round (and all between if not there)
+    // updates current_round and initializes the vectors corresponding to the given round (and all between if not there)
     fn init_round(&mut self, round: Round) {
+        if self.current_round < round {
+            self.current_round = round;
```

Comment:
This is the only place, where we modify `current_round`, and the `current_round` is always equal to one less than the length of `candidates_by_round`. Can we replace the `current_round` with a method calculating it from `candidates_by_round`?

Reply:
Done